cmd/cli/sign.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sah4ez/pspk/pkg/keys"
+	"github.com/sah4ez/pspk/pkg/utils"
+	"github.com/urfave/cli"
+	"strings"
+)
+
+func Sign() cli.Command {
+	return cli.Command{
+		Name:        "sign",
+		Aliases:     []string{"s"},
+		Description: "Signs the message through ed25519",
+		Usage:       "--name <KEY_NAME> sign <MESSAGE>",
+		Action: func(c *cli.Context) error {
+			message := c.Args()[0:]
+			name := c.GlobalString("name")
+			if name == "" {
+				if cfg.CurrentName == "" {
+					return errors.New("empty current name, set to config or use --name")
+				}
+				name = cfg.CurrentName
+			}
+
+			path = path + "/" + name
+
+			priv, err := utils.Read(path, "key.bin")
+			if err != nil {
+				return errors.Wrap(err, "can not read key.bin")
+			}
+
+			privArray := utils.Slice2Array32(priv)
+
+			sign := keys.Sign(&privArray, []byte(strings.Join(message, " ")), utils.Random())
```

Comment:
You used `keys.Sign` but I didn't see this changes in PR. :-)

Reply:
I changed that part I am using the private key for signature and key.Sign
```
priv, err := utils.Read(path, "key.bin")
			if err != nil {
				return errors.Wrap(err, "can not read key.bin")
			}

			privArray := utils.Slice2Array32(priv)

			sign := keys.Sign(&privArray, []byte(strings.Join(message, " ")), utils.Random())
			data := base64.StdEncoding.EncodeToString(sign[:])
```

Is it correct ?

Reply:
Yes, it's correct. 
But I didn't see changes in the `key` package, there should be `Sign` and `verify` methods.

Reply:
I am using from https://github.com/sah4ez/pspk/blob/06fbc8d0d2d2f2925e30bdc9afe731701c0e3fb8/pkg/keys/keys.go

Reply:
could you try it my branch?

Reply:
Oh... I forgot that I added it xD
That all ok, I'll merge it today. Thanks!

Reply:
Oh... I forgot that I added it xD
That all ok, I'll merge it today. Thanks!

Reply:
I forgot the part of --file in the command `sign --file <path_to_file>`
could be in another PR?
and could you give an example about  `sign --file <path_to_file>`
Thanks!

Reply:
Yes, you can create another PR with support `--file` flag.